sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/MesosResource.java

```diff
@@ -56,27 +56,23 @@ public Value getValue() {
     }
 
     public String getRole() {
-        if (Capabilities.getInstance().supportsPreReservedResources()) {
+        if (resource.getReservationsCount() > 0) {
             return getRefinedRole();
         } else {
             return getLegacyRole();
         }
     }
 
     private String getRefinedRole() {
-        if (resource.getReservationsCount() > 0) {
-            return resource.getReservations(resource.getReservationsCount() - 1).getRole();
-        }
-
-        return Constants.ANY_ROLE;
+        return resource.getReservations(resource.getReservationsCount() - 1).getRole();
```

Comment:
in the `0` case we should still return `ANY_ROLE` to avoid relying on legacy behavior.  The role returned by `getLegacyRole` will not be a field for much longer.